soc/xtensa/intel_s1000/memory.h

```diff
@@ -43,9 +43,16 @@
 #define MEM_ERROR_TEXT_SIZE			0x180
 #define MEM_ERROR_LIT_SIZE			0x8
 
-/* text and data share the same L2 HP SRAM on Intel S1000 */
-#define TEXT_BASE				(L2_SRAM_BASE + L2_VECTOR_SIZE)
-#define TEXT_SIZE				0x16000
+/* text and data share the same L2 HP SRAM on Intel S1000.
+ * So, they lie next to each other.
+ */
+#define RAM_BASE				(L2_SRAM_BASE + L2_VECTOR_SIZE)
+#define RAM_SIZE				0x2E000
```

Comment:
Make `RAM_SIZE` as large as possible. Can be 4MB - memory need for vector tables and space allocated for stacks.

Reply:
Done.

Reply:
You set it to 4MB which is not correct. 4MB is the total memory, out of which `L2_VECTOR_SIZE` is taken and I think the stack the grows from the highest address to lower addresses, so some space need to be reserved for the stack.
Also, the 4MB is defined in DTS and duplicated as `L2_SRAM_SIZE` use the macro instead of hardcoded number.

Reply:
The L2_VECTOR_SIZE definitely needs to be subtracted but I am not sure we need to take into consideration the stack. I looked at all other socs for Xtensa in zephyr and the memory region for code, data and bss extend till the end. __stack is defined right at the end of the memory. So, it is possible that the __stack may end up overlapping with the main sections but i guess it is left to the App developers to ensure that doesn't happen. Also, in Zephyr, i think only some libraries may use this stack and mostly be empty for most applications. So, if we set hard limits we may just be simply wasting the memory. @nashif can comment more on this.

Reply:
@nashif : Can you please comment on how __stack gets used? You can look at the linker.ld of S1000 to see the sections that are put under this.

Reply:
@andrewboie : Can you please comment on how __stack gets used? You can look at the linker.ld of S1000 to see the sections that are put under this.

Reply:
+1 for the question from @rgundi to @andrewboie